Support bundling and Java-side requires

hdeshev opened the pull request:
Webpack bundling will split the final bundle in two chunks:

- tns-java-classes.js: containing NativeScriptApplication, NativeScriptActivity, and all related code.
- bundle.js: containing the rest of the library and application code.

We need to load tns-java-classes (if present) earlier, so that Java-JavaScript mappings get registered.

[Here's](https://github.com/NativeScript/nativescript-dev-webpack/blob/master/tns-java-classes.js) a sample tns-java-classes.js *before* webpacking -- those `require` calls will get changed by webpack.

Ping @jasssonpet, @KristinaKoeva

blagoev commented:
looks ok. 
Can we do this using Javascript only. Maybe changing the ts_helpers.js file to require the app/tns-java-classes.js if it exists? 

Or maybe we can change the runtime to run every module in internal dir on startup and place the tns-java-classes.js there.

hdeshev commented:
@blagoev I moved the `tns-java-classes.js` load code to `ts_helpers.js`, but that feels worse than the original setup:

- I had to change RuntimeHelper to use `runModule` instead of `runScript` for `ts_helpers.js`
- The `ts_helpers.js` code doesn't have app dir information, and I had to take it from `__dirname` (the reason I switched to `runModule` above.
- The code is a bit more complex.

I suggest I drop the last commit and merge the Java (RuntimeHelper) fix I initially proposed.

jasssonpet commented:
👍 This would be required for the snapshot plugin, too.

atanasovg commented:
I think the initial proposal looks better. Ideally, we would do what @blagoev suggests - to traverse the internal directory and run every JS file found there. But given the short time frame  before the release I suggest we just merge @hdeshev initial proposal - :+1: from me for this.

hdeshev commented:
Reverted to the initial implementation. Merge at will. :angel:

blagoev commented:
:+1 for the traverse we may need ordering, so lets just keep this for now
